chainio/clients/elcontracts/reader.go

```diff
@@ -60,29 +67,63 @@ type ELReader interface {
 }
 
 type ELChainReader struct {
-	logger            logging.Logger
-	elContractsClient clients.ELContractsClient
-	ethClient         eth.EthClient
+	logger                  logging.Logger
+	slasher                 slasher.ContractSlasherCalls
+	delegationManager       delegationmanager.ContractDelegationManagerCalls
+	strategyManager         strategymanager.ContractStrategyManagerCalls
+	blsPubkeyCompendium     blspkcompendium.ContractBLSPublicKeyCompendiumCalls
+	blsPubKeyCompendiumAddr common.Address
+	ethClient               eth.EthClient
 }
 
 // forces EthReader to implement the chainio.Reader interface
 var _ ELReader = (*ELChainReader)(nil)
 
 func NewELChainReader(
-	elContractsClient clients.ELContractsClient,
+	slasher slasher.ContractSlasherCalls,
+	delegationManager delegationmanager.ContractDelegationManagerCalls,
+	strategyManager strategymanager.ContractStrategyManagerCalls,
+	blsPubkeyCompendium blspkcompendium.ContractBLSPublicKeyCompendiumCalls,
+	blsPubKeyCompendiumAddr common.Address,
 	logger logging.Logger,
 	ethClient eth.EthClient,
-) (*ELChainReader, error) {
+) *ELChainReader {
 	return &ELChainReader{
-		elContractsClient: elContractsClient,
-		logger:            logger,
-		ethClient:         ethClient,
-	}, nil
+		slasher:                 slasher,
+		delegationManager:       delegationManager,
+		strategyManager:         strategyManager,
+		blsPubkeyCompendium:     blsPubkeyCompendium,
+		blsPubKeyCompendiumAddr: blsPubKeyCompendiumAddr,
+		logger:                  logger,
+		ethClient:               ethClient,
+	}
+}
+
+func BuildELChainReader(
+	slasherAddr gethcommon.Address,
+	blsPubKeyCompendiumAddr gethcommon.Address,
+	ethClient eth.EthClient,
+	logger logging.Logger,
+) (*ELChainReader, error) {
+	elContractBindings, err := chainioutils.NewEigenlayerContractBindings(slasherAddr, blsPubKeyCompendiumAddr, ethClient, logger)
+	if err != nil {
+		return nil, err
+	}
+	return NewELChainReader(
+		elContractBindings.Slasher,
+		elContractBindings.DelegationManager,
+		elContractBindings.StrategyManager,
+		elContractBindings.BlsPubkeyCompendium,
+		blsPubKeyCompendiumAddr,
+		logger,
+		ethClient,
+	), nil
 }
 
+// TODO(samlaf): should we just pass the CallOpts directly as argument instead of the context?
```

Comment:
the reason we don't want it - what if over time abigen upgrades and deprecate this CallOpts for something else. Our interface will break - in current we hide all the dependencies interface and objects so that probability of breaking changes is low

Reply:
But geth has been around for much longer, so their interface is less likely to change frequently, whereas if we start doing this, its our interface that is going to be breaking all the time (oh this method we need more control on startBlock, let's add it to interface, etc.). Eventually we'll just end up with the geth interface on most functions.

Reply:
We agreed to make the changes. Updated to new api in [6e2c821](https://github.com/Layr-Labs/eigensdk-go/pull/73/commits/6e2c8217737086857aab8f15824fed3ce0d4688d) and [d999091](https://github.com/Layr-Labs/eigensdk-go/pull/73/commits/d9990916f28f96e754bf7674cfbb620faa5f1f2c)